e2e/test/clusterautoscaler/provider.go

```diff
@@ -0,0 +1,58 @@
+// +build k8srequired
+
+package clusterautoscaler
+
+import (
+	"github.com/giantswarm/apiextensions/v2/pkg/clientset/versioned"
+	azureclient "github.com/giantswarm/e2eclients/azure"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type ProviderConfig struct {
+	AzureClient *azureclient.Client
+	G8sClient   versioned.Interface
+	Logger      micrologger.Logger
+
+	ClusterID  string
+	CtrlClient client.Client
+}
+
+type Provider struct {
+	azureClient *azureclient.Client
+	g8sClient   versioned.Interface
+	logger      micrologger.Logger
+
+	clusterID  string
+	ctrlClient client.Client
+}
+
+func NewProvider(config ProviderConfig) (*Provider, error) {
+	if config.AzureClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.AzureClient must not be empty", config)
+	}
+	if config.G8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.G8sClient must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+	if config.ClusterID == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.ClusterID must not be empty", config)
+	}
+	if config.CtrlClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.CtrlClient must not be empty", config)
+	}
+
+	p := &Provider{
+		azureClient: config.AzureClient,
```

Comment:
I believe you don't need `azureClient`, `clusterID` or `ctrlClient`.

Reply:
removed